### DISCUSSION TITLE
Management UI returns JS errors on specific pages/nodes

### DISCUSSION THREAD

**Discussion by Miscki** [rabbitmq/rabbitmq-server#12999] [Questions] [answered]
### Describe the bug

Just started learning how to use and deploy RabbitMQ so it might be a mistake on my part but i cant think of anything more to check.

- 3 node cluster
- deployed by puppet
- management plugin enabled
- users, vhost and user permissions created by puppet (2 users, 1vhost, 1 user permission)

When i login to the web ui on node1 or node3 i cannot see exchanges, queues or in the admin section - users, policies or limits.
Those however all work fine on node2.
I have checked users, vhosts, exchanges and user permissions locally on the nodes using rabbitmqctl and they appear to be the same.

The error i get is 
`ReferenceError: ac is not defined ReferenceError: ac is not defined at Array.process (eval at compile (http://rabbitmqnode3:15672/js/ejs-1.0.min.js:1:6654), :140:2) at EJS.render (http://rabbitmqnode3:15672/js/ejs-1.0.min.js:1:1885) at format (http://rabbitmqnode3:15672/js/main.js:1154:21) at http://rabbitmqnode3:15672/js/main.js:486:24 at with_reqs (http://rabbitmqnode3:15672/js/main.js:1136:9) at http://rabbitmqnode3:15672/js/main.js:1132:17 at req.onreadystatechange (http://rabbitmqnode3:15672/js/main.js:1227:17)` 

I have also tested to confirm but i can send messages to that vhost from any node and receive from any node so the cluster appears to function as intended, only the web ui seems to have problems.

### Reproduction steps

1. Deploy with puppet
2. 1st node runs puppet waiting for other nodes to start before creating the cluster
3. 2nd node runs puppet forming the cluster and adding users 
4. only the 2nd node can view those resources

### Expected behavior

The web UI should work the same on all nodes in a cluster. Right?

**Comment by lukebakken** [comment, marked as answer]
Hello, thanks for using RabbitMQ.

The issue you report is not eligible for free community support from Team RabbitMQ ([support policy](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md#who-is-eligible-for-community-support)). In addition, Team RabbitMQ does _not_ maintain anything Puppet-related.

It's _extremely_ unlikely that you have found a bug with RabbitMQ or the Management UI. More likely, you have an issue in your specific environment, or, there is something wrong with Puppet (also unlikely).

I suggest you do the following:
* Open an issue [here](https://github.com/voxpupuli/puppet-rabbitmq)
* Provide _CODE_ to deploy RabbitMQ using Puppet the exact same way you are. Right now, you are asking someone to take time to set up an environment the same way as you and you don't even provide whatever code/configuration is necessary to set something up in the same way.

All someone should have to do to reproduce the issue is the following:
```sh
git clone https://github.com/Miscki/rabbitmq-puppet-issue.git
cd rabbitmq-puppet-issue
make # or ./setup.sh, or some Puppet command
```

Anything more than the above and you are asking someone to spend far too much time giving you free support.

**Reply by Miscki** [reply]
Apologies for the lack of puppet code, did not think of that, i will follow up with an issue with the module maintainers as recommended. Thanks

**Comment by michaelklishin** [comment]
@Miscki you haven't even bothered to mention what RabbitMQ version is used.

With a probability of 0.99, you have an existing browser cache that the [Upgrade docs](https://www.rabbitmq.com/docs/upgrade#rabbitmq-plugins-compatibility) explicitly require you to clear.

As of `4.0.5`, clearing the cache after upgrades should be not necessary in at least some cases due to https://github.com/rabbitmq/rabbitmq-server/pull/12779.

**Reply by Miscki** [reply]
It is the most recent version indeed, 4.0.5 freshly deployed, not upgraded.

**Reply by michaelklishin** [reply]
An absolute majority of users do not run into this exception and those that do almost always had assets from earlier versions cached.

`4.0.5` introduces asset caching-related headers but it cannot clear all caches and local data for you.

After clearing all local caches and data associated with the management UI domain, use Developer Tools to see if some files might have failed to load (there really isn't a lot of them, though) and such.

**Reply by Miscki** [reply]
Seems like you are right, after clearing the browser cache from using an older deployment the issue resolved. Thanks a lot.
